pkg/syncer/storagepool/migrationController.go

```diff
@@ -147,7 +147,12 @@ func (m *migrationController) migrateVolume(ctx context.Context, pvc *unstructur
 	}
 	log.Debugf("Migrating volume %v to SP %v", volumeID, targetSP.GetName())
 
-	err = m.relocateCNSVolume(ctx, volumeID, targetSPName)
+	relocateFn := func() error {
+		return m.relocateCNSVolume(ctx, volumeID, targetSPName)
+	}
+	initBackoff := time.Duration(100) * time.Millisecond
+	maxBackoff := time.Duration(60) * time.Second
+	err = RetryOnError(relocateFn, initBackoff, maxBackoff, 1.5)
```

Comment:
Subbu, thanks a lot for the diff. Just a small comment. RetryOnError function will retry on and on till the func succeeds. We can add another field called `numRetries` or `retries` to RetryOnError and pass it to `ExponentialBackoff` function instead of `math.MaxInt32`.

Reply:
Thanks for catching this. Introduced `maxRetries` as a param. Also manually verified that with the selected value `maxRetries=16` and `initBackoff=100ms` it takes about 2 mins of retries before giving up. That should be sufficient time to wait for transient connectivity issues with VC to get sorted out.